src/mono/mono/eventpipe/ep-rt-mono-runtime-provider.c

```diff
@@ -1554,8 +1554,15 @@ bulk_type_log_single_type (
 	// Sets val variable sized parameter type data, type_parameters_count, and mono_type_parameters associated
 	// with arrays or generics to be recursively batched in the same ep_rt_mono_log_type_and_parameters call
 	switch (mono_underlying_type->type) {
-	case MONO_TYPE_ARRAY:
 	case MONO_TYPE_SZARRAY:
+	{
+		// FIXME: Previously was handled by below ARRAY block but that is incorrect; this implementation is speculative -kg
```

Comment:
I assume you should be able to handle both cases similar, except that you can't assume mono_type is MonoArrayType it could also be MonoClass, so you would just need to handle the case where you get the type, rank and mono_type_parameter differently.